src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.volasoftware.tinder.services.implementations;
+
+import com.volasoftware.tinder.constants.AccountConstant;
+import com.volasoftware.tinder.exceptions.AccountNotFoundException;
+import com.volasoftware.tinder.exceptions.FriendExistException;
+import com.volasoftware.tinder.exceptions.FriendNotFoundException;
+import com.volasoftware.tinder.models.Account;
+import com.volasoftware.tinder.repositories.AccountRepository;
+import com.volasoftware.tinder.services.contracts.FriendService;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class FriendServiceImpl implements FriendService {
+
+    private final AccountRepository accountRepository;
+
+    @Override
+    public void addFriend(Long friendId) {
+        Account loggedAccount = getLoggedAccount();
+        Account friendAccount = getAccountById(friendId);
+
+        if (hasFriends(loggedAccount)) {
+            if (friendExist(loggedAccount, friendAccount)) {
+                throw new FriendExistException(AccountConstant.ALREADY_EXIST);
```

Comment:
When this occur will show "Account already exist". Can you make the message more clear

Reply:
Message was updated.

Reply:
To be more helpful, can you add id or any unique field for the user and the friend?